connection.go

```diff
@@ -1167,10 +1167,14 @@ func read(r io.Reader, lenbuf []byte) (response []byte, err error) {
 		err = errors.New("wrong response header")
 		return
 	}
-	length = (int(lenbuf[1]) << 24) +
-		(int(lenbuf[2]) << 16) +
-		(int(lenbuf[3]) << 8) +
-		int(lenbuf[4])
+	length = (uint64(lenbuf[1]) << 24) +
+		(uint64(lenbuf[2]) << 16) +
+		(uint64(lenbuf[3]) << 8) +
+		uint64(lenbuf[4])
+	if length > math.MaxUint32 {
+		err = errors.New("response is too big")
+		return
+	}
```

Comment:
It would be more clear to have a separate length-check block:
```suggestion
	switch {
	case length == 0:
		err = errors.New("response is too big")
		return
	case length > math.MaxUint32:
		err = errors.New("response should not be 0 length")
		return
	}
```